menas/ui/generic/functions.js

```diff
@@ -53,5 +53,15 @@ var Functions = new function () {
         }
       }
     })
+  };
+
+  /**
+   * Checks if the property is enum-like by making sure that non-empty inerable of `allowedValues` exists
+   * @param propertyType
+   * @returns {boolean}
+   */
+  this.isEnum = function(propertyType) {
+    return propertyType.hasOwnProperty("allowedValues") && propertyType.allowedValues.length > 0;
```

Comment:
What if it's an enum without allowed values (`allowedValues.length == 0`)?

Reply:
I like both suggestions 👍

Reply:
Done. 
 - `isEnum` -> `hasValidAllowedValues` ✔️ 
 - enum property type(s) the allowedValues: does not allow "" to be present + must have 1+ items ✔️